app-backend/db/src/main/scala/com/azavea/rf/database/UserGroupRoleDao.scala

```diff
@@ -91,17 +94,36 @@ object UserGroupRoleDao extends Dao[UserGroupRole] {
           .filter(fr"is_active = true")
           .selectOption
       }
+      roleTargetEmail <- UserDao.unsafeGetUserById(subjectId) map { _.email }
+      roleCreatorEmail <- UserDao.unsafeGetUserById(actingUser.id) map { _.email }
       existingMembershipStatus = existingRoleO map { _.membershipStatus }
+      rolesMatch = existingRoleO.map(
+        (ugr: UserGroupRole) => ugr.groupRole == userGroupRoleCreate.groupRole
+      ).getOrElse(false)
       createdOrReturned <- {
-        (existingMembershipStatus, adminCheck) match {
-          case (Some(MembershipStatus.Requested), true) | (Some(MembershipStatus.Invited), false) =>
+        (existingMembershipStatus, adminCheck, rolesMatch) match {
+          // Only admins can change group roles, and only approved roles can have their group role changed
+          case (Some(MembershipStatus.Approved), true, false) =>
+            UserGroupRoleDao.create(userGroupRoleCreate.toUserGroupRole(actingUser, MembershipStatus.Approved)) <*
+              UserGroupRoleDao.deactivate(existingRoleO.map( _.id ).get, actingUser)
+          // Accepting a role requires agreement about what the groupRole should be -- users can't cheat
+          // and become admins by accepting a MEMBER role by posting an ADMIN role
+          case (Some(MembershipStatus.Requested), true, true) | (Some(MembershipStatus.Invited), false, true) =>
             UserGroupRoleDao.create(userGroupRoleCreate.toUserGroupRole(actingUser, MembershipStatus.Approved)) <*
               UserGroupRoleDao.deactivate(existingRoleO.map( _.id).get, actingUser)
-          case (None, true) =>
-            UserGroupRoleDao.create(userGroupRoleCreate.toUserGroupRole(actingUser, MembershipStatus.Invited))
-          case (None, false) =>
-            UserGroupRoleDao.create(userGroupRoleCreate.toUserGroupRole(actingUser, MembershipStatus.Requested))
-          case (Some(_), _) => existingRoleO.get.pure[ConnectionIO]
+          // rolesMatch will always be false when existingRoleO is None, so don't bother checking it
+          case (None, true, _) =>
+            UserGroupRoleDao.create(userGroupRoleCreate.toUserGroupRole(actingUser, MembershipStatus.Invited)) <*
```

Comment:
There are at least two ways this can fail --

The parameters can be bad (smtp port is negative, something else), in which case we'll throw either an `IllegalArgumentException` (negative smtp port, e.g.) or an `EmailException` (bad email addresses, e.g.). These are handled in `Email.scala`

Sending the email can fail despite good parameters -- this would happen if all of the parameters were valid, but the email address didn't exist or we lost network for a minute or something I think. This is handled in `Notify.scala`